planetcantile/data/generate.py

```diff
@@ -41,42 +44,44 @@ def __post_init__(self):
-gcs_mars_2000_sphere = CRS(f"""GEOGCRS["GCS_Mars_2000_Sphere",{gcs_mars_2000_sphere_datum},CS[ellipsoidal,2],AXIS["latitude",north,ORDER[1],ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9122]]],AXIS["longitude",east,ORDER[2],ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9122]]]]""")
-
-moon_2000_ellipsoid = 'SPHEROID["Moon_2000_IAU_IAG",1737400.0,0.0,LENGTHUNIT["metre",1]]'
-moon_2000_datum = f'DATUM["D_Moon_2000_Sphere",{moon_2000_ellipsoid},PRIMEM["Greenwich",0],UNIT["Decimal_Degree",0.0174532925199433]]'
-moon_2000_sphere = CRS(f'GEOGCS["Moon_2000",{moon_2000_datum},CS[ellipsoidal,2],AXIS["latitude",north,ORDER[1],ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9122]]],AXIS["longitude",east,ORDER[2],ANGLEUNIT["degree",0.0174532925199433,ID["EPSG",9122]]]]')
-
-# uber tuple of all TmsMatrixSets to generate, if you want something new add it here
-crss = (
-    Tmsparam(
-        crs=gcs_mars_2000_sphere,
-        extent=(-180.0, -90.0, 180.0, 90.0),
-        title="GCS_Mars_2000_Sphere TileMatrixSet",
-        identifier="GCS_Mars_2000_Sphere",
-        maxzoom=22,  # set lower for 200mpp MOLA
-        geographic_crs=gcs_mars_2000_sphere
-    ),
-    Tmsparam(
-        crs=moon_2000_sphere,
-        extent=(-180.0, -90.0, 180.0, 90.0),
-        title="Moon_2000_Sphere TileMatrixSet",
-        identifier="Moon_2000_Sphere",
-        maxzoom=22,
-        geographic_crs=moon_2000_sphere
-    ),
-
+# Parse all the GeoCRSs from the list
+geocrss = []
+for wkt_str in resp.split(os.linesep + os.linesep):
+    if 'GEOGCRS' in wkt_str[:7] and 'TRIAXIAL' not in wkt_str:  # insanely hacky
+        geocrss.append(wkt_str)
 
-)
+# Build a dynamic crss list
+crss = []
+for crs in geocrss:
+    crs_obj = CRS(crs)
+    title = crs_obj.name
+    auth = crs_obj.to_authority(min_confidence=25)
+    if auth is not None:
+        authority_version, code = auth
+        authority, version = authority_version.split('_')
+        identifier = f'{authority}_{code}_{version}'
+        
+        tmsp = Tmsparam(
+            crs=crs_obj,
+            extent=(-180.0, -90.0, 180.0, 90.0),  # Hardcoded domains, ever differ?
+            title=title,
+            identifier=identifier,
+            maxzoom=22,  # set lower for 200mpp MOLA
+            geographic_crs=crs_obj
+        )
+        crss.append(tmsp)
+    else:
+        print(f'Could not find authority for {crs_obj.to_wkt()}')
 
 for tmsp in crss:
     # create the tms object
     tms = morecantile.TileMatrixSet.custom(**asdict(tmsp))
     _d = json.loads(tms.json(exclude_none=True))  # get to pretty printed json
-    with open(f'./{tmsp.crs.name}_tms.json', 'w') as dst:
+    with open(f'./{tmsp.identifier}_tms.json', 'w') as dst:
         # dump to json
         json.dump(_d, dst, indent=4, ensure_ascii=False)
         print(f'wrote {dst.name}')
```

Comment:
So this `generate` script will create a lot of TMS right? I'm kinda worry now that when importing `planetcantile` we might try to parse too many JSON files before being able to do any operation.

https://github.com/AndrewAnnex/planetcantile/blob/5ea2577f5dc4a3bc91b0443ef0633a5f89b15e03/planetcantile/defaults.py#L10


Maybe this could be fixed on morecantile side where we only `parse` the JSON file when we need it 🤷

Reply:
I did start a PR to show how it could be https://github.com/developmentseed/morecantile/pull/94

I don't think this is too much of a change. in planetcantile it will look like:

```python
import importlib.resources as importlib_resources
from morecantile.defaults import default_tms, TileMatrixSets

planet_tms_jsons = importlib_resources.files('planetcantile.data').glob('*.json')
default_tms.update(
    {p.stem: p for p in planet_tms_jsons}
)
tms = TileMatrixSets(copy(default_tms))

# Or to only use TMS from planetcantile
import importlib.resources as importlib_resources
from morecantile import TileMatrixSets

planet_tms_jsons = importlib_resources.files('planetcantile.data').glob('*.json')
default_tms = {p.stem: p for p in planet_tms_jsons}
tms = TileMatrixSets(copy(default_tms))
```

The other solution would be to simply use `TILEMATRIXSET_DIRECTORY` environment variable to tell morecantile to look for json document within a specific directory. https://github.com/developmentseed/morecantile/blob/main/morecantile/defaults.py#L16-L18

Reply:
> We could enforce filename mapping to IDs and just read the filenames (this has issues).

Well right now morecantile assume this 🙈 

> I guess, I would really want to make sure that we can advertise the availability of the TMS while still maintain good performance? Maybe I'm worried about the wrong thing though!

Right now (with the PR) we keep the `advertising` part (but using the file name).

Reply:
> Well right now morecantile assume this 🙈
😆 This is good! And we can use the environment variable on the titiler side to offer these via our lambda deploy. I might hack the check to opengis.net out of a version that I have deployed (only to be able to test until they update) and then get that URL shared to you both for verification.

Knowing about that env variable and having the code snippet is AWESOME! 🌟

Reply:
> Knowing about that env variable

Yeah I should document this :-)

Reply:
Oh and feel free to comment directly on the morecantile PR 🙏

Reply:
@vincentsarago I commented on your PR and think the lazy loading is a good change to make both here and in morecantile. I do wonder a bit about the environment variable option however and I wonder if there could be a way to keep using the importlib_resources pattern to allow planetcantile to become an extension of morecantile without needing to explicitly import it on the user side, or given an import by the user monkey-patch/update the morecantile's tilematrixsets object. One could imagine setting up similar extensions to make tmss for other authorities.

Reply:
well we specifically designed `morecantile.defaults.tms` to not be overwritten (we use `copy(default_tms)`) 

I see the appeal of being able to quickly extend morecantile default tms but on the other hand I think writting 
```python
import importlib.resources as importlib_resources
from morecantile.defaults import default_tms, TileMatrixSets

planet_tms_jsons = importlib_resources.files('planetcantile.data').glob('*.json')
default_tms.update(
    {p.stem: p for p in planet_tms_jsons}
)
tms = TileMatrixSets(copy(default_tms))
``` 
is not that bad 🤷‍♂️

Reply:
I am worried about projects that use morecantile's default tms set without allowing overrides. Yes it's possible to make PRs against them to accept a tms set as a parameter but my suggestion could be a very cool mechanism to support

Reply:
err "like" it but not exactly it, also not working currently

Reply:
> I am worried about projects that use morecantile's default tms set without allowing overrides. 

I'm not sure to understand this 

Morecantile is a tool that lets you define and use TMS grid, having a set of defaults in the project is a bonus but isn't the goal of the project. 

> Yes it's possible to make PRs against them to accept a tms set as a parameter but my suggestion could be a very cool mechanism to support

I'm open for suggestion but right now I have no idea how it could look 🤷

Reply:
> err "like" it but not exactly it, also not working currently

what is not working?